File: CR.Elastos-Nucleus/service/sample_code/python/watch_eth_contract.py
import json

from elastos_adenine.sidechain_eth import SidechainEth

def main():
    host = "localhost"
    port = 8001
    production = False

    api_key = "9A5Fy8jDxsJQSDdU4thLZs9fwDmtVzBU"
    did = "iHdasfhasdflkHdasfasdfD"
    network = "gmunet"
    smart_contract_address = "0xdf29327c95b12A37089A6e230d5ce50F23237671"
    smart_contract_name = "HelloWorld"
    smart_contract_code_hash = "QmRCn3tQem7UugGLE7tkchXudp4prqLtDhMRs828mUED34"

    try:
        sidechain_eth = SidechainEth(host, port, production)
        print("\n--> Watch ETH Contract")
        response = sidechain_eth.watch_eth_contract(api_key, did, network, smart_contract_address, smart_contract_name,
                                                    smart_contract_code_hash)
        if response['status']:
            json_output = json.loads(response['output'])
            print("Status Message :", response['status_message'])
            for i in json_output['result']:
                print(i, ':', json_output['result'][i])
        else:
            print("Error Message: " + response['status_message'])
    except Exception as e:
        print(e)
    finally:
        sidechain_eth.close()

if __name__ == '__main__':
    main()

File: CR.Elastos-Nucleus/service/sample_code/go/watch_eth_contract.go
package main

import (
	"encoding/json"
	"log"

	"github.com/cyber-republic/go-grpc-adenine/elastosadenine"
)

func main() {
    grpcServerHost := "localhost"
    grpcServerPort := "8001"
    production := false
    
    apiKey := "9A5Fy8jDxsJQSDdU4thLZs9fwDmtVzBU"
    network := "gmunet"
    smartContractAddress := "0xdf29327c95b12A37089A6e230d5ce50F23237671"
    smartContractName := "HelloWorld"
    smartContractCodeHash := "QmRCn3tQem7UugGLE7tkchXudp4prqLtDhMRs828mUED34"
    
    log.Println("--> Watch ETH Contract")
    sidechainEth := elastosadenine.NewSidechainEth(grpcServerHost, grpcServerPort, production)
    defer sidechainEth.Close()
    response := sidechainEth.WatchEthContract(apiKey, network, smartContractAddress, smartContractName, smartContractCodeHash)
    if response.Output != "" {
    	output := []byte(response.Output)
    	var jsonOutput map[string]interface{}
    	json.Unmarshal(output, &jsonOutput)
    	log.Printf("Status Message : %s", response.StatusMessage)
    	result, _ := json.Marshal(jsonOutput["result"].(map[string]interface{}))
    	log.Printf(string(result))
    }
}
